modal/_asgi.py

```diff
@@ -1,20 +1,25 @@
 # Copyright Modal Labs 2022
 import asyncio
-from typing import Any, Callable, Dict
+from typing import Any, AsyncGenerator, Callable, Dict, List
 
 from modal_utils.async_utils import TaskContext
 
 from ._blob_utils import MAX_OBJECT_SIZE_BYTES
+from .config import logger
 from .functions import current_function_call_id
 
 
-def asgi_app_wrapper(asgi_app, function_io_manager):
+class NoRequestReceived(Exception):
+    pass
```

Comment:
Could we be consistent about placing exceptions in `modal.exception`?

Reply:
You might be right, and I think some of my other changes here might make this unnecessary

Reply:
Managed to get rid of it in the followup PR